### PR TITLE
ARM: added VMSS support for NIC list & show cmdlets

### PR DESCRIPTION
**Pull request by romangromov**
This PR contains minor changes related to `network nic list/show` in `ARM`:
* added support for Virtual machine Scale Set while listing or showing NICs
* `azure-arm-network` package replaced with local file **temporary** because it's not pushed yet to `NPM`.

This changes requested with high priority by @DeepakRajendranMsft and @FrankSiegemund.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@romangromov__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by romangromov** [comment]
@yugangw-msft @amarzavery about test cases failure: this issue not related to this PR and came from previously merged PR https://github.com/Azure/azure-xplat-cli/pull/2164, please see Travis logs https://travis-ci.org/Azure/azure-xplat-cli/jobs/83047750

**Comment by yugangw-msft** [comment]
@romangromov, this is a known test flakiness which has going on for a while which  because the public template used by tests appear has changed. Thanks for the opening this and we will fix the test in more reliable way. For now, I will disable that test.

**Comment by romangromov** [comment]
@yugangw-msft PR synced with `dev`, please merge

File: lib/util/utils.js [hunk 1]
```diff
@@ -181,7 +181,7 @@ exports.createComputeResourceProviderClient = function (subscription) {
 };
 
 exports.createNetworkResourceProviderClient = function (subscription) {
-  var factoryMethod = require('azure-arm-network').createNetworkResourceProviderClient;
+  var factoryMethod = require('./../commands/arm/network/sdk/network').createNetworkResourceProviderClient;
```
Comment (yugangw-msft) [inline lib/util/utils.js]: We should not reference this temp file, please ask network team to publish a new version of 'azure-arm-network'. Let me know if you don't know the people to contact

File: lib/commands/arm/network/nic._js [hunk 2]
```diff
@@ -163,11 +167,22 @@ __.extend(Nic.prototype, {
 
   show: function (resourceGroupName, nicName, options, _) {
     var self = this;
-    var nic = self.get(resourceGroupName, nicName, _);
+    var nic = null;
+
+    if (options.virtualMachineScaleSet || options.virtualMachineIndex) {
+      if (!(options.virtualMachineScaleSet && options.virtualMachineIndex)) {
```
Comment (yugangw-msft) [inline lib/commands/arm/network/nic._js]: nit,the code might be more readable if gets organized like the following, up to you though.
if (options.virtualMachineScaleSet && options.virtualMachineIndex) {

} else if (options.virtualMachineScaleSet || options.virtualMachineIndex) {
  //error
} else {
}

**Comment by yugangw-msft** [comment]
Please talk to network team to publish a new version of network node library.
The rest looks good to me.

**Comment by yugangw-msft** [comment]
@DeepakRajendranMsft, please publish the node sdk for npm

**Comment by DeepakRajendranMsft** [comment]
I have published the package now. we just got the SDK merged

**Comment by romangromov** [comment]
@yugangw-msft @DeepakRajendranMsft thanks to all, SDK local files removed now, `azure-arm-network` package updated to latest version from npm

File: lib/commands/arm/network/nic._js (hunk 2 above)
Comment (romangromov) [inline lib/commands/arm/network/nic._js]: thanks, I'm good with existing expression

**Pull request merged**